docs/concepts/bitswap.md

```diff
@@ -11,4 +10,54 @@ related:
+
+### Session
+
+IPFS content blocks are often connected to each other through a [Merkle DAG](https://docs-beta.ipfs.io/concepts/merkle-dag/). If a sender's block requests are related, a mechanism called a _Bitswap session_ can optimize block requests sent to other peers. This mechanism can increase transfer speed and reduce the number of duplicate blocks on the network. Active peers are favored, as they are tracked relative to them having the requested blocks and how quickly they respond.
+
+### Provider/Providing
+
+The provider system is responsible for announcing and reannouncing to the IPFS network that a node has content. If a node cannot find content with connected peers, it will query a content routing system (usually a form of [DHT](https://docs-beta.ipfs.io/concepts/dht/)) to locate a peer with the content. Bitswap routes these requests through the `ProviderQueryManager` system. As a node receives blocks, it announces them on the provided content routing system.
+
+### Deeper Examination
+
+A diagram showing the primary tasks of Bitswap and underlying components:
+
+![Bitswap task and component diagram](https://github.com/ipfs/go-bitswap/blob/master/docs/go-bitswap.png "Go-Bitswap")
+
+#### Message Queue
+
+An IPFS node has a message queue for each active peer. The message queue has the next message to be sent to that peer. Updates to a message queue come from either the wantlist manager or the decision engine. Task workers watch the message queues and send queued messages to the associated peers.
+
+#### Want-Manager
+
+The want-manager handles our requests for blocks. When CIDs are added or removed from our wantlist, these updates must be sent to connected peers. This data is sent to all relevant peers' message queues. Once a node receives a block that it asked for, it should send out a notification called a 'Cancel' to tell its peers that it no longer requires those blocks.
+
+#### Decision Engine
+
+The decision engine processes incoming messages that contain a wantlist. It checks the CID for each block in the wantlist against the local block-store and creates a task in the peer request queue for each block it finds. Task workers pull the tasks off the queue, retrieve the block requested, and send it off.
+
+The decision engine makes use of a ledger as a record of the aggregate data exchanged between two peers. An IPFS node maintains one ledger for each connected peer.
+
+### Additional references
+
+[Article: Swapping bits and distributing hashes on the decentralized web (Textile)](https://medium.com/textileio/swapping-bits-and-distributing-hashes-on-the-decentralized-web-5da98a3507)
+
+"About Bitswap" poster from the IPFS developer summit in Berlin in July 2018:
+!['About Bitswap' poster](https://user-images.githubusercontent.com/74178/43230914-f818dab2-901e-11e8-876b-73ba6a084f76.jpg "Bitswap-Poster_Berlin-July-2018")
```

Comment:
This image needs to be updated and recreated in Photoshop/Illustrator at some point. That's outside the scope of this bounty though.

Reply:
@johnnymatthews You might want to file an issue in https://github.com/protocol/design-shop/ for that. Not sure where it'll fit into Agata's overall timeline, but probably worth getting it into her queue.